src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -243,7 +250,7 @@ public static bool HasDisposeSignatureByConvention(this IMethodSymbol method)
         /// </summary>
         public static bool HasDisposeBoolMethodSignature(this IMethodSymbol method)
         {
-            if (method.Name == "Dispose" && method.MethodKind == MethodKind.Ordinary &&
+            if (method.IsImplicitOrExplicitInterfaceImplementationName("Dispose", "System.IDisposable") &&
```

Comment:
Can we add a test for `Dispose(bool)` explicit implementation case?

Reply:
Same suggestion here and below, use `WellKnownTypeNames.SystemIDisposable` instead of hard coded string.